[18.0] [CHG] analytic_brand: replace analytic account by analytic distribution

marielejeune opened the pull request:
Depends on https://github.com/OCA/brand/pull/231

sbejaoui committed:
[ADD] - Add analytic brand module

sbejaoui committed:
[ADD] - Add analytic_account_id field

sbejaoui committed:
[IMP] - set the invoice line analytic account at brand change

oca-travis committed:
[UPD] Update analytic_brand.pot

[UPD] README.rst

OCA-git-bot committed:
analytic_brand 12.0.1.0.1

sbejaoui committed:
[MIG] - migration analytic_brand 13.0

oca-travis committed:
[UPD] Update analytic_brand.pot

[UPD] README.rst

[FIX] Fixed travis.

AmmarOfficewalaSerpentCS committed:
[MIG][14.0] Migarted analytic_brand module.

[UPD] Update analytic_brand.pot

sbejaoui committed:
[15.0][MIG] - analytic_brand

Commit:
[UPD] Update analytic_brand.pot

OCA-git-bot committed:
[UPD] README.rst

BT-tkarpinski committed:
[FIX] brand view in analytic_brand

OCA-git-bot committed:
analytic_brand 15.0.1.0.1

OCA-git-bot committed:
[UPD] README.rst

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (2 of 2 strings)

Translation: brand-15.0/brand-15.0-analytic_brand
Translate-URL: https://translation.odoo-community.org/projects/brand-15-0/brand-15-0-analytic_brand/es/

JasminSForgeFlow committed:
[IMP] analytic_brand: black, isort, prettier

JasminSForgeFlow committed:
[MIG] analytic_brand: Migration to 17.0

Commit:
[UPD] Update analytic_brand.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (2 of 2 strings)

Translation: brand-17.0/brand-17.0-analytic_brand
Translate-URL: https://translation.odoo-community.org/projects/brand-17-0/brand-17-0-analytic_brand/it/

BhaveshHeliconia committed:
[MIG] analytic_brand: Migration to 18.0

marielejeune committed:
[CHG] analytic_brand: replace analytic account by analytic distribution

samirGuesmi commented on analytic_brand/tests/test_analytic_brand.py:
```diff
@@ -0,0 +1,39 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo.addons.base.tests.common import BaseCommon
+
+
```
from odoo.tests.common import tagged

@tagged("post_install", "-at_install")

marielejeune replied:
Done in a new commit with a detailed explication in the commit message.

samirGuesmi reviewed (commented):
If possible, please tag the tests to avoid errors caused by missing required fields added by other modules.

marielejeune committed:
[FIX] analytic_brand: Run tests in post-install:

There is a common issue in v18 when creating a new res.partner in tests
(which is the case here as we create a res.brand).
There is a required field autopost_bills in account addon.
As this addon doesn't depend on account, if we install this addon
on a DB where account is also installed, tests will fail because
the autopost_bills column is created into postgres, with the not null constraint,
but the default value won't be set by Odoo.

Running these tests in post-install ensure Odoo will set the default value
as specified in the account addon.

yvaucher commented:
/ocabot merge nobump
